torch/autograd/profiler_util.py

```diff
@@ -271,6 +271,12 @@ def supported_export_stacks_metrics(self):
         ]
 
     def export_stacks(self, path: str, metric: str):
+        # Support passing more meaningful names for privateuse1 users
+        metric = (
+            metric.replace(self._use_device, "privateuse1")
```

Comment:
should _use_device be more generic, since we may have other devices in the future?

Reply:
Without this replacing, metric only can be set to "self_privateuse1_time_total" for privateuse1 backend. but our user may pass a more meaningful name.
For example, our privateuse1 backend named "MLU", so self._use_device will be set to "mlu".  
We prefer to set metric to  "self_mlu_time_total", there will raise an error if no replacing. 
Of course, we can also use "self_privateuse1_time_total" , in which case replacing will do nothing.

For other devices, their self._user_device may be some ohter strings,  they can use the original string "self_privateuse1_time_total" as metric and also can use self_{self._use_device}_time_total. The logical is generic.

So, i believe self._use_device should be specific, like the current design, allowing it to represent different device names. Additionally, we can use self._use_device  to extend some more generic strings like self_privateuse1_time_total to specific.

Reply:
Hi, @aaronenyeshi @briancoutinho , could you please give some suggestions about this?

Reply:
Okay, i will revert this.


```python
                    metric_value = getattr(
                        evt,
                        metric.replace("cuda", "device")
                        .replace("xpu", "device")
                        .replace("privateuse1", "device"),
                    )
```
By the way, i noticed the above code in https://github.com/pytorch/pytorch/pull/123247/files,   Actually, this modification doesn't solve the issue I wanted to address. What I'm looking to do is something like appending a new line `.replace(self._use_device, "device")`. However, this isn't a critical issue, just something for convenience.  For now, I'll remove my modification.